src/charm.py

```diff
@@ -330,26 +330,29 @@ def create_kubeconfigs(self):
         node_name = self.get_node_name()
         public_server = self.k8s_api_endpoints.external()
 
-        if not os.path.exists("/root/.kube/config"):
+        if not os.path.exists(ROOT_KUBECONFIG):
             # Create a bootstrap client config. This initial config will allow
             # us to get and create auth webhook tokens via the Kubernetes API,
             # but will not have the final admin token just yet.
-            kubernetes_snaps.create_kubeconfig(
-                "/root/.kube/config",
-                ca=ca,
-                server=local_server,
-                user="admin",
-                token=auth_webhook.token_generator(),
-            )
+            token = auth_webhook.token_generator()
+        else:
+            token = None
+
+        kubernetes_snaps.update_kubeconfig(
+            ROOT_KUBECONFIG,
+            ca=ca,
+            server=local_server,
+            token=token,
+            user="admin",
+        )
```

Comment:
the crux of the fix is to ensure the ca is updated in the kubeconfig

Reply:
ok - so this works because we may not have the api server up yet, hence can't verify `kubectl get secrets`, and if the CA changes during this time, our "bootstrap" client config wasn't getting updated because we were checking for "if file dne".

is that scenario what you're fixing with this?

Reply:
Right, according to the issue tracked in the bug, the certificate was invalid in the stack trace starting at 

```
auth_webhook.create_token(...)
   auth_webhook.get_token(...)
       auth_webhook.get_secrets(...)
            kubectl(...)
```

I surmised that the CA could be updated here to ensure we always have the right cert in place before checking the tokens.